pipx 0.13.0.1 (new formula)

AlJohri opened the pull request:
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

Homebrew formula for [pipx](https://github.com/pipxproject/pipx)- allows installing python CLI tools in isolated virtual environments.

zbeekman commented:
Aren't there other packages that do this?

AlJohri commented:
@zbeekman This project replaces [pipsi](https://github.com/mitsuhiko/pipsi). I am not aware of any other package that does this.

pipx alleviates issues when using pyenv, asdf, etc. as your CLI tools will remain unaffected as python versions change.

zbeekman commented:
I was thinking of pipsi. Thanks! LGTM!

zbeekman commented on Formula/pipx.rb:
```diff
@@ -0,0 +1,28 @@
+  end
+
+  test do
+    assert_match "PIPX_HOME", shell_output("#{bin}/pipx --help")
```
Anyway we can get better tests to ensure everything is working correctly? Like installing a simple, light weight pip script/program, testing it, and uninstalling it?

Or anything else we can do beyond just testing the output of `--help`?

AlJohri replied:
Sure thing! Will add.

zbeekman replied:
Can you please update tests to be more thorough and amend and force push? I'd love to see this merged soon.

AlJohri replied:
Done! (https://github.com/Homebrew/homebrew-core/pull/39411)

zbeekman commented on Formula/pipx.rb:
```diff
@@ -0,0 +1,28 @@
+    Run this command to ensure ~/.local/bin is on your PATH:
+      pipx ensurepath
+  EOS
+  end
```
Also, I think the caveats block should go too; this advice is not specific to homebrew installation of pipx, AFAICT. Caveats are for homebrew specific stuff.

AlJohri replied:
@zbeekman I think this is actually fairly specific to `pipx` since adding this to the PATH is not necessarily required for normal python usage. Many people never use `pip install --user xyz` and thus they never need to have a path such as `~/.local/bin` in their PATH.

For `pipx` in particular, it will simply not work without this command being run. It's the first recommended step here in the installation instructions: https://github.com/pipxproject/pipx#install-pipx

What do you think?

zbeekman replied:
@Homebrew/maintainers: thoughts on this caveats block? I think it could go either way...

vitorgalvao replied:
Agreed on removing the `caveat` because it’s not HB-specific. That’s the approach we take on casks, but I don’t know how prevalent that is on formulae.

fxcoudert replied:
Agreed: we don't want `caveats` to turn into general documentation.

AlJohri replied:
Sounds good- will remove.

AlJohri commented:
@zbeekman I made some mistake in force pushing and had to open a new PR: https://github.com/Homebrew/homebrew-core/pull/39411

I've fixed up the test and removed the caveats. Can you take a look?
